SocialCareCaseViewerApi.Tests/V1/UseCase/AllocationsUseCaseTests.cs

```diff
@@ -78,12 +78,97 @@ public void ListAllocationsByWorkerEmail()
             var request = new ListAllocationsRequest() { WorkerEmail = "test@example.com" };
             var gatewayResponse = new List<Allocation> { new Allocation() { AllocatedWorker = "Test Worker" } };
 
-            _mockDatabaseGateway.Setup(x => x.SelectAllocations(0, 0, "test@example.com"))
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(0, 0, "test@example.com", 0))
                 .Returns(gatewayResponse);
 
             var response = _allocationsUseCase.Execute(request);
 
             response.Allocations.Should().BeEquivalentTo(gatewayResponse);
         }
+
+        [Test]
+        public void ListAllocationsByTeamIdReturnsExpectedList()
+        {
+            var request = new ListAllocationsRequest() { TeamId = 10 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(0, 0, null, 10))
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeEquivalentTo(gatewayResponse);
+        }
+
+        [Test]
+        public void ListAllocationsByTeamIdReturnsNothingIfNoMatches()
+        {
+            var request = new ListAllocationsRequest() { TeamId = 8 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(0, 0, null, 10))
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeNullOrEmpty();
+        }
+
+        [Test]
+        public void ListAllocationsByMosaicIdReturnsExpectedList()
+        {
+            var request = new ListAllocationsRequest() { MosaicId = 3 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(3, 0, null, 0))
+                .Returns(gatewayResponse);
+
+            var response = _allocationsUseCase.Execute(request);
+
+            response.Allocations.Should().BeEquivalentTo(gatewayResponse);
+        }
+
+        [Test]
+        public void ListAllocationsByMosaicIdReturnsNothingIfNoMatches()
+        {
+            var request = new ListAllocationsRequest() { MosaicId = 3 };
+            var gatewayResponse = new List<Allocation> { new Allocation() { Id = 1, PersonId = 2, AllocatedWorker = "Test Worker" } };
+
+            _mockDatabaseGateway.Setup(x => x.SelectAllocations(3, 0, null, 3))
```

Comment:
As previously mentioned this is redundant. gatewayResponse and setup in this case can be removed.